### DISCUSSION TITLE
Next.js starter does not (cannot?) deploy on Cloudflare on Free Tier

### DISCUSSION THREAD

**Discussion by btahir** [cloudflare/workers-sdk#7030] [Q&A] [answered]
Hi - Frankly, I'm not sure if this is a bug or intended behavior. The default Next.js app starter does not deploy on Cloudflare Workers because it hits the 1MB limit on the free tier. However, this seems ridiculously low, especially considering that the paid tier only increases the limit to 10MB. The starter app is 1.5MB compressed (gzip).

I thought the KV setup was meant to offload some of this bloat, but either that's not happening or this is simply a limitation of Workers. I'd like to understand what's possible with Cloudflare Workers and Next.js and whether this truly is the limit.

I'm just following the instruction to deploy the starter app [here](https://developers.cloudflare.com/workers/frameworks/framework-guides/nextjs/).

Thanks!

**Comment by vicb** [comment]
@btahir as we mention [in the docs](https://opennext.js.org/cloudflare#cloudflare), the current package is quite experimental. We should be at a much better place in the coming weeks. We are aware of the [size issue](https://github.com/opennextjs/opennextjs-cloudflare/issues/77) and will work on it once we stabilize the package.

I hope this helps

**Comment by emily-shen** [comment, marked as answer]
Hey, just to update this thread - we've raised the limit for free tier to 3MB, so this should work now 🎉
https://developers.cloudflare.com/workers/platform/limits/#account-plan-limits
feel free to open an issue if you encounter any other problems :)

**Comment by btahir** [comment]
Thanks!

**Comment by david-ng-hk** [comment]
Hi @vicb and @emily-shen 
I test the same app https://developers.cloudflare.com/workers/frameworks/framework-guides/nextjs/ today
it can't deploy on the free plan, I guess it is the upload size over 3M (Total Upload: 4780.80 KiB / gzip: 1209.07 KiB)

**Reply by vicb** [reply]
3MiB is gzipped so you still have some headroom
